modules/custom/activity_creator/activity_creator.module

```diff
@@ -255,6 +256,15 @@ function _activity_creator_activity_for_entity_updater($action, EntityInterface
   if ($entity !== NULL) {
     $entity_type = $entity->getEntityTypeId();
     $entity_id = $entity->id();
+
+    // We never create activities for configuration entities but accessing
+    // entity queries during configuration changes may cause errors.
+    if ($entity instanceof ConfigEntityInterface) {
+      return;
+    }
+
+    // TODO: The if statement below can be refactored after the code that was
```

Comment:
I've been looking at this particular piece of code too for a few times already... Did you already think about how to fix this? IMO I think we should remove all of those entityQuery and just use the entitytype manager to load the related activities. Thoughts?

Reply:
I'm not entirely sure how the entityTypeManager code would look. I don't think that's necessarily a good idea because we don't really need to entities themselves, we only need the IDs to invalidate them (unless we can get the entity IDs of the related entities without loading the actual entities, then that would probably save us a roundtrip to the database altogether because the entity we're fetching related entities for is already loaded?).

I think moving to a switch can already make this more readable, or even ensuring we don't potentially run queries multiple times for the user, group and group_content types.

Something like:

```php
// Never update activities as a result of an activity update.
if ($entity_type === 'activity') {
  return;
}

$ids = [];
switch ($entity_type) {
  case 'user':
  case 'group':
    $entity_query = \Drupal::entityQuery('activity');
    $entity_query->condition('field_activity_recipient_' . $entity_type, $entity_id, '=');
    $ids = $entity_query->execute();
    break;
  case 'group_content':
    /** @var \Drupal\group\Entity\GroupContent $group_content */
    $group_content = $entity;
    $linked_entity = $group_content->getEntity();
    $group = $group_content->getGroup();
    if ($linked_entity && $linked_entity->getEntityTypeId() === 'node' && $group->id()) {
      $entity_query = \Drupal::entityQuery('activity');
      $entity_query->condition('field_activity_entity.target_id', $linked_entity->id(), '=');
      $entity_query->condition('field_activity_entity.target_type', $linked_entity->getEntityTypeId(), '=');
      $entity_query->condition('field_activity_recipient_group', $group->id(), '=');
      $ids = $entity_query->execute();
    }
    break;
  default:
    $entity_query = \Drupal::entityQuery('activity');
    $entity_query->condition('field_activity_entity.target_id', $entity_id, '=');
    $entity_query->condition('field_activity_entity.target_type', $entity_type, '=');
    $ids = $entity_query->execute();
}
```

While we're at it, we can also replace
```php
$tags = [];
foreach ($ids as $id) {
  $tags[] = 'activity:' . $id;
}
Cache::invalidateTags($tags);
```

With
```
$tags = array_map(
  function ($id) { return "activity:${id}"; }, 
  $ids
);
Cache::invalidateTags($tags);
```

Reply:
I like the code... however, keep in mind it's not just about fetching the IDs of the activities, since there's also a (deprecated) call to `entity_delete_multiple` at the bottom.

Reply:
I think we should tackle these things in a follow-up issue (hence the TODO). The change in this PR fixes a pretty serious bug, the code below can use some refactoring (and Drupal 9 proofing) but otherwise still works.